### PR TITLE
Image Editor

### PR DESCRIPTION
**Pull request by cadecairos**
https://webmademovies.lighthouseapp.com/projects/65733/tickets/1806-create-editor-for-image-plugin

### CODE REVIEW

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @cadecairos received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/dc48a5823d19e4d/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/dc48a5823d19e4d/output.txt

Total script time: 0.82 mins

Server is running on http://pr763.butyr.org/templates/basic/
Run the Test Suite: http://pr763.butyr.org/test/

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @cadecairos received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/6baa5a828884037/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/6baa5a828884037/output.txt

Total script time: 1.11 mins

Server is running on http://pr763.butyr.org/templates/basic/
Run the Test Suite: http://pr763.butyr.org/test/

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @cadecairos received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/6f6cfe7c0b32e39/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/6f6cfe7c0b32e39/output.txt

Total script time: 0.85 mins

Server is running on http://pr763.butyr.org/templates/basic/
Run the Test Suite: http://pr763.butyr.org/test/

File: templates/assets/editors/image/popcorn.image.editor.html [hunk 1]
```diff
@@ -0,0 +1,79 @@
+<div class="butter-editor image-editor">
+  <h1>Image Event Editor</h1>
+  <div class="error-message-container">
+    <div class="error-message"></div>
+  </div>
+  <div class="editor-options-wrapper">
+      <div class="editor-options">
```
Comment (k88hudson) [inline templates/assets/editors/image/popcorn.image.editor.html]: Too much space!

File: templates/assets/editors/image/popcorn.image.editor.html [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+          <input type="number" id="editor-out-value" class="editor-text-input"/>
+
+          <div class="editor-header">
+            Source
+          </div>
+
+          <div class="editor-radio-wrapper">
+            <input type="radio" name="source-type" disabled="true" id="editor-dragdrop-radio" class="editor-input-radio" />
+            <div class="editor-radio-label">
+                You can drop an image from your desktop on the container
```
Comment (k88hudson) [inline templates/assets/editors/image/popcorn.image.editor.html]: Too much space

File: templates/assets/editors/image/popcorn.image.editor.html [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+      </div>
+  </div>
+</div>
+
+<style type="text/css">
+  .butter-editor.image-editor .editor-options-wrapper {
+    top: 40px;
+  }
+  .error-message-container{
+
```
Comment (k88hudson) [inline templates/assets/editors/image/popcorn.image.editor.html]: Remove

File: templates/assets/editors/image/popcorn.image.editor.js [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+      _inputOut = _rootElement.querySelector( "input#editor-out-value" ),
+      _dragDropRadio = _rootElement.querySelector( "input#editor-dragdrop-radio" ),
+      _urlRadio = _rootElement.querySelector( "input#editor-url-radio" ),
+      _flickrRadio = _rootElement.querySelector( "input#editor-flickr-radio" ),
+      _urlInput = _rootElement.querySelector( "input#editor-url" ),
+      _tagInput = _rootElement.querySelector( "input#editor-flickr-tag" ),
+      _galleryUrlInput = _rootElement.querySelector( "input#editor-flickr-url" ),
+      _numberInput = _rootElement.querySelector( "input#editor-flickr-number" ),
+      _transitionIn = _rootElement.querySelector( "select#editor-in-select" ),
+      _transitionOut = _rootElement.querySelector( "select#editor-out-select" ),
```
Comment (k88hudson) [inline templates/assets/editors/image/popcorn.image.editor.js]: Please switch these all to classes

File: templates/assets/editors/image/popcorn.image.editor.js [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+    _dragDropRadio.checked = true;
+  }
+
+  function enableUrl() {
+    if ( _flickrActive ) {
+      disableFlickr( true );
+    } else if ( _dragDropActive ) {
+      disableDragDrop();
+    }
+    disableUrl( false )
```
Comment (k88hudson) [inline templates/assets/editors/image/popcorn.image.editor.js]: semicolon

File: templates/assets/plugins/image/popcorn.image.js [hunk 6]
```diff
@@ -234,23 +244,23 @@
...
           elem: "input",
           type: "number",
           label: "Flickr: Count",
           optional: true
         },
+        width: {
+          elem: "input",
+          type: "number",
+          label: "Width",
+          "default": 80
```
Comment (k88hudson) [inline templates/assets/plugins/image/popcorn.image.js]: Add `"units": "%"` please

File: templates/assets/plugins/image/popcorn.image.js [hunk 7]
```diff
@@ -234,23 +244,23 @@
...
+        },
         height: {
           elem: "input",
           type: "number",
           label: "Height",
-          "default": 100
+          "default": 80
```
Comment (k88hudson) [inline templates/assets/plugins/image/popcorn.image.js]: same thing here

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @cadecairos received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/d1698422a694398/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/d1698422a694398/output.txt

Total script time: 0.79 mins

Server is running on http://pr763.butyr.org/templates/basic/
Run the Test Suite: http://pr763.butyr.org/test/

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @cadecairos received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/5dedb420c235494/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/5dedb420c235494/output.txt

Total script time: 1.03 mins

Server is running on http://pr763.butyr.org/templates/basic/
Run the Test Suite: http://pr763.butyr.org/test/

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @cadecairos received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/fa7b898f80b2c7a/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @cadecairos received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/a9d3248a6239ca0/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/fa7b898f80b2c7a/output.txt

Total script time: 1.29 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/a9d3248a6239ca0/output.txt

Total script time: 0.77 mins

Server is running on http://pr763.butyr.org/templates/basic/
Run the Test Suite: http://pr763.butyr.org/test/

**Comment by jbuck** [comment]
Staged on master: https://github.com/mozilla/butter/commit/e8c4364b2456e16f0c8dabb11856f9508e94f502

**Pull request closed without merging**